src/replication.c

```diff
@@ -986,14 +1205,14 @@ read_io_resp(spec_t *spec, replica_t *replica, read_event_t *revent)
 
 			*read_count = 0;
 			if (resp_hdr->len != 0) {
-			/* this will not be NULL for mgmt_ack IO */
 				if ((*resp_data) == NULL)
 					(*resp_data) = malloc(resp_hdr->len);
 			}
-			*state = READ_IO_RESP_DATA;
```

Comment:
this place is correct right for setting the state..

Reply:
ok. I was thinking do we need to check for errorno after read_data as read_data itself will give count lesser than reqlen if read_data hit any error during read operation.